Food_Tracker_App/app/src/main/java/com/example/foodtracker/ui/recipes/RecipeDisplay.java

```diff
@@ -26,4 +107,15 @@ private void createNavbar() {
                 .replace(R.id.recipe_display_nav_bar, navBar)
                 .commit();
     }
+
+    /**
+     * Instantiates the top bar fragment for the recipe display menu
+     */
+    private void createTopBar() {
+        TopBar topBar = TopBar.newInstance("Recipe Details", false);
+        getSupportFragmentManager().beginTransaction()
+                .setReorderingAllowed(true)
+                .replace(R.id.topBarContainerView, topBar)
+                .commit();
+    }
```

Comment:
I changed the top bar fragment so that if the activity is recipeDisplay it will go back to recipesMainScreen.

Reply:
Sounds good, it looks fine to me but the only thing I can think of is that maybe we want this logic to be more generic. Two things I can think of:

1. Have a boolean that says that we do not want to return to the main screen, i.e. `finishActivity` that we pass on the initialization of the top bar and which defaults to false.. or even an enum i.e. `BackButtonBehavior` that has like `RETURN_TO_MAIN_MENU` and `FINISH_ACTIVITY`

2. If we wanted it to be even more generic we could add another method to the top bar listener and send a callback optionally to do instead of returning to the main menu.

The reason I like this better is just because it makes it so that the navbar doesn't need to know who is using it!

Reply:
Good call! I added a finishActivity boolean to newInstance() in TopBar - if this is not what you had in mind let me know!